### DISCUSSION TITLE
Can not start helloworld demo for arm rk3568 on ubuntu 18.04

### DISCUSSION THREAD

**Discussion by zrz4066** [slint-ui/slint#3399] [Q&A]
I can not start the  hello world slint demo,The following are the error message.

thread 'main' panicked at 'Failed to initialize any backend! Wayland status: NoCompositorListening X11 status: LibraryOpenError(OpenError { kind: Library, detail: "opening library failed (Dynamic loading not supported); opening library failed (Dynamic loading not supported)" })', /home/ice/.cargo/registry/src/rsproxy.cn-8f6827c7555bfaf8/winit-0.28.6/src/platform_impl/linux/mod.rs:757:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtraceroot@linux:~#

     .-/+oossssoo+/-.               root@linux
        `:+ssssssssssssssssss+:`           ----------
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 18.04.5 LTS aarch64
    .ossssssssssssssssssdMMMNysssso.       Host: rpdzkj dr4-rk3568 base on Rockchip RK3568 EVB1 DDR4 V10 Board
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 4.19.232
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 4 mins
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 1332
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 4.4.20
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1920x1080
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   Theme: Adwaita [GTK3]
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   Icons: Adwaita [GTK3]
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Terminal: /dev/pts/1
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   CPU: (4) @ 1.992GHz
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Memory: 325MiB / 3900MiB
  +sssssssssdmydMMMMMMMMddddyssssssss+
   /ssssssssssshdmNNNNmyNMMMMhssssss/
    .ossssssssssssssssssdMMMNysssso.
      -+sssssssssssssssssyyyssss+-
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.

**Comment by ogoffart** [comment]
What happens is that Slint could not open neither a wayland nor an X11 connection. Are you running any of these?

The error to open an X11 connection is 
>  opening library failed (Dynamic loading not supported)

Which is curious. Are you using musl by any chance?   I think winit will need to load some x11 dynamic library to work.

The error to open wayland is 
> NoCompositorListening

Which basically mean that no wayland server is running.


(I converted the issue into a discussion because it is more suited for such questions while the issue tracker is for bug tracking)

**Comment by zrz4066** [comment]
I use musl 。 I use the following command to generate  executable file
cargo build --release  --target armv7-unknown-linux-musleabihf

The following is  error message of the runnning demo 

![Screenshot  running slint demo](https://github.com/slint-ui/slint/assets/16775484/067b7846-4c61-4ec5-87bd-799bb84ba5dc)

**Reply by ogoffart** [reply]
I don't know if winit with x11 can be supported.
Apparently it can if you add crt-static in your flags: https://github.com/rust-windowing/winit/issues/1818#issuecomment-903765478
